src/submission/validation-clinical/followUp.ts

```diff
@@ -24,13 +24,82 @@ import {
 	SubmissionValidationOutput,
 	DonorVitalStatusValues,
 } from '../submission-entities';
-import { ClinicalEntitySchemaNames, FollowupFieldsEnum } from '../../common-model/entities';
+import {
+	ClinicalEntitySchemaNames,
+	DonorFieldsEnum,
+	FollowupFieldsEnum,
+} from '../../common-model/entities';
 import { DeepReadonly } from 'deep-freeze';
 import { Donor, Treatment } from '../../clinical/clinical-entities';
 import _ from 'lodash';
 import { getClinicalEntitiesFromDonorBySchemaName } from '../../common-model/functions';
 import { getEntitySubmitterIdFieldName } from '../../common-model/functions';
 import * as utils from './utils';
+import { checkForExceptions } from '../exceptions/exceptions';
+
+const checkDonorTimeConflict = (
+	donorDataToValidateWith: { [k: string]: any },
+	followUpRecord: DeepReadonly<SubmittedClinicalRecord>,
+	errors: SubmissionValidationError[],
+) => {
+	if (
+		donorDataToValidateWith.donorVitalStatus === DonorVitalStatusValues.deceased &&
+		donorDataToValidateWith.donorSurvivalTime <
+			followUpRecord[FollowupFieldsEnum.interval_of_followup]
+	) {
+		errors.push(
+			utils.buildSubmissionError(
+				followUpRecord,
+				DataValidationErrors.FOLLOW_UP_DONOR_TIME_CONFLICT,
+				FollowupFieldsEnum.interval_of_followup,
+				{},
+			),
+		);
+	}
+};
+
+function checkTreatmentTimeConflict(
+	followUpRecord: DeepReadonly<SubmittedClinicalRecord>,
+	treatment: DeepReadonly<Treatment>,
+	errors: SubmissionValidationError[],
+) {
+	// A follow up may or may not be associated with treatment
+	if (treatment == undefined) return;
+
+	if (
+		followUpRecord.interval_of_followup &&
+		treatment.clinicalInfo &&
+		treatment.clinicalInfo.treatment_start_interval &&
+		followUpRecord.interval_of_followup <= treatment.clinicalInfo.treatment_start_interval
+	) {
+		errors.push(
+			utils.buildSubmissionError(
+				followUpRecord,
+				DataValidationErrors.FOLLOW_UP_CONFLICING_INTERVAL,
+				FollowupFieldsEnum.interval_of_followup,
+				[],
+			),
+		);
+	}
+}
+
+function getExistingFollowUp(
+	existingDonor: DeepReadonly<Donor>,
+	record: DeepReadonly<SubmittedClinicalRecord>,
+) {
+	if (existingDonor.followUps) {
+		return getClinicalEntitiesFromDonorBySchemaName(
+			existingDonor,
+			ClinicalEntitySchemaNames.FOLLOW_UP,
+		).find(
+			(ci) =>
```

Comment:
Legacy function but opportunity to improve variable name `ci` here

Reply:
Good call. Will leave as is for now because this is just a bug fix PR rather than a chore or feature in our terminology